service/service

```diff
@@ -142,6 +144,10 @@ _d_upgrade () {
 
 	# run any database migrations/updates needed
 	sudo -u $PHAB_USER $ROOT/phabricator/bin/storage upgrade --force
+
+    # clear out old backups
+    cd $BAKPATH
+    ls -tp | grep -v revlog | tail -n +$(( $MAX_BACKUPS + 1)) | xargs -d '\n' rm -rf --
```

Comment:
1. Correct
2. Correct. The -v flag allows you to filter out lines that match a particular phrase.
3. Basically, the -n +# flags combine to say "list all elements at or after this point." I've chosen n+1 so that tail lists everything other than the n most recent copies.
4. Correct. This particular element of the pipe is specific to GNU xargs, which is more flexible than the BSD version available on OSX.